src/main/java/nl/hsac/fitnesse/fixture/util/HttpClient.java

```diff
@@ -1,35 +1,73 @@
 package nl.hsac.fitnesse.fixture.util;
 
-import org.apache.http.Header;
-import org.apache.http.HeaderElement;
-import org.apache.http.HttpEntity;
-import org.apache.http.NameValuePair;
+import org.apache.http.*;
 import org.apache.http.client.CookieStore;
-import org.apache.http.client.methods.HttpDelete;
-import org.apache.http.client.methods.HttpGet;
-import org.apache.http.client.methods.HttpPost;
-import org.apache.http.client.methods.HttpPut;
-import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.methods.*;
 import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.client.utils.URIBuilder;
 import org.apache.http.entity.ContentType;
 import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.DefaultRedirectStrategy;
 import org.apache.http.impl.client.HttpClients;
 import org.apache.http.protocol.BasicHttpContext;
 import org.apache.http.protocol.HttpContext;
 import org.apache.http.util.EntityUtils;
+import org.apache.http.util.TextUtils;
 
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Locale;
 import java.util.Map;
-
 /**
  * Helper to make Http calls and get response.
  */
 public class HttpClient {
     private final static org.apache.http.client.HttpClient HTTP_CLIENT;
 
+    public static boolean followRedirect;
+
     static {
-        HTTP_CLIENT = HttpClients.custom().useSystemProperties().disableContentCompression()
-                .setUserAgent(HttpClient.class.getName()).build();
+
+        HTTP_CLIENT = HttpClients.custom()
+                .useSystemProperties()
+                .disableContentCompression()
+                .setUserAgent(HttpClient.class.getName())
+                .setRedirectStrategy(new DefaultRedirectStrategy() {
+                    @Override public boolean isRedirected(HttpRequest request, org.apache.http.HttpResponse response, HttpContext context) throws ProtocolException {
+                        boolean isRedirect = super.isRedirected(request, response, context);
+                        if(isRedirect && !followRedirect) {
```

Comment:
To me this if statement is not very clear. It probably is the name of the superclass' method, or the local variable.

Can you may make a bit clearer (by naming or comments) what the various cases are/mean?

Furthermore I would prefer it if the method only has a single return statement. I find that makes it a bit easier to understand the flow.